app/controllers/trainee/subjects_controller.rb

```diff
@@ -1,8 +1,35 @@
 class Trainee::SubjectsController < TraineesController
-  before_action :authenticate_user!, :trainee?, :store_location, only: :show
+  before_action :authenticate_user!, :store_location, :load_course,
+                :load_subject_by_course, :load_course_subject, only: :show
 
   def show
-    @course = Course.find_by id: params[:course_id] if params[:course_id]
-    @subject = Subject.find_by id: params[:id] if params[:id]
+    @user_course_subject = @course_subject.user_course_subjects
+                                          .find_by user_id: current_user.id
+  end
+
+  private
+
+  def load_course
+    @course = Course.find_by id: params[:course_id]
+    return if @course
+
+    flash[:error] = t "notice.error"
+    redirect_to trainee_courses_path
```

Comment:
lặp nhiều quá
giờ đổi sang dùng find hết cho c
xong tham khảo cách làm chỗ 13 vote ấy
https://stackoverflow.com/questions/2385799/how-to-redirect-to-a-404-in-rails

Reply:
dùng find bắn ra exception thì mình bắt exception
render ra trang 404
k phải xử lý redirect nữa

Reply:
dạ vâng ạ